salt/_modules/metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import absolute_import, unicode_literals, print_function
+import base64
+import errno
+import logging
+import os
+import tempfile
+
+from salt.exceptions import CommandExecutionError
+from salt.ext.six import iteritems
+from salt.ext import six
+import salt.utils.files
+import salt.utils.platform
+import salt.utils.templates
+import salt.utils.versions
+import salt.utils.yaml
+from salt.exceptions import TimeoutError
+from salt.ext.six.moves import range  # pylint: disable=import-error
+
+
+# from salt.exceptions import CommandExecutionError
+# import salt.utils.files
+# import salt.utils.versions
+# from salt.ext.six.moves import range  # pylint: disable=import-error
+
+
+try:
+    import kubernetes
+    import kubernetes.client
+
+    HAS_LIBS = True
+except ImportError:
+    HAS_LIBS = False
+
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'metalk8s_kubernetes'
```

Comment:
The fact that we use the same virtual name to merge methods is not obvious. I'd like a module docstring that mentions it, please :)